distribution/pull_v2_windows.go

```diff
@@ -23,16 +23,17 @@ func (ld *v2LayerDescriptor) Descriptor() distribution.Descriptor {
 }
 
 func (ld *v2LayerDescriptor) open(ctx context.Context) (distribution.ReadSeekCloser, error) {
-	if len(ld.src.URLs) == 0 {
-		blobs := ld.repo.Blobs(ctx)
-		return blobs.Open(ctx, ld.digest)
-	}
-
 	var (
 		err error
 		rsc distribution.ReadSeekCloser
 	)
 
+	// Try to get the layer from the registry.
+	blobs := ld.repo.Blobs(ctx)
+	if rsc, err = blobs.Open(ctx, ld.digest); err == nil {
+		return rsc, err
```

Comment:
I don't think `Open` will return an error if the blob does not exist. The error would be returned after the first read. So I expect this would always return here and prevent pulling from the foreign URL.